tools/bridge_inundation.py

```diff
@@ -0,0 +1,127 @@
+import argparse
+import errno
+import os
+from timeit import default_timer as timer
+
+import geopandas as gpd
+import pandas as pd
+
+
+def bridge_risk_status(hydrofabric_dir: str, flow_file_dir: str, output_dir: str) -> gpd.GeoDataFrame:
+    """
+    This function detect which bridge points are affected by a specified flow file. The function requires a flow file (expected to follow
+    the schema used by 'inundation_mosaic_wrapper') with data organized by 'feature_id' and 'discharge' in cms. The output includes a geopackage
+    containing bridge points labeled as "threatened", "at risk", or "not at risk" based on forcasted discharge compared to preset discharge
+    ("max_discharge" or "max_discharge75").
+
+    Args:
+        hydrofabric_dir (str):    Path to hydrofabric directory where FIM outputs were written by
+                                    fim_pipeline.
+        flow_file_dir (str):      Path to flow file to be used for inundation.
+                                    feature_ids in flow_file should be present in supplied HUC.
+        output (str):             Path to output geopackage.
+    """
+
+    dir_path = hydrofabric_dir
+    # Check that hydrofabric_dir exists
+    if not os.path.exists(dir_path):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), dir_path)
+
+    # Get the list of all hucs in the directory
+    entries = os.listdir(dir_path)
+    hucs = []
+    for entry in entries:
+        # create the full path of the entry
+        full_path = os.path.join(dir_path, entry)
+        # check if the netry is a directory
+        if os.path.isdir(full_path):
+            hucs.append(entry)
+
+    # Check that flow file exists
+    if not os.path.exists(flow_file_dir):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), flow_file_dir)
+
+    # Read the flow_file
+    dtype_dict = {'feature_id': str}
+    flow_file = pd.read_csv(flow_file_dir, dtype=dtype_dict)
+
+    # Initialize an empty list to hold GeoDataFrames
+    gdfs = []
+    # Iterate through hucs
+    for huc in hucs:
+        print(f'Processing HUC: {huc}')
+        # Construct the file path
+        gpkg_path = os.path.join(dir_path, huc, 'osm_bridge_centroids.gpkg')
+        # Check if the file exists
+        if not os.path.exists(gpkg_path):
+            print(f"No GeoPackage file found in {gpkg_path}. Skipping...")
+            continue
+        # Open the bridge point GeoPackage for each huc
+        bri_po = gpd.read_file(gpkg_path)
```

Comment:
What happens if there are no bridges for a HUC? Does that mean the `osm_bridge_centroids.gpkg` doesn't exist (as opposed to an empty GPKG file) and the script exits correctly?

Reply:
I checked `data/clip_vectors_to_wbd.py` and `osm_bridge_centroids.gpkg` won't be written if no bridges are found so this isn't an issue.